### ISSUE TITLE
Override UI templates

### ISSUE THREAD

**Issue by davidsneighbour** [CloudCannon/pagefind#288]
Is there any way to override `pagefind_ui/default/svelte/ui.svelte` or parts of it? In my concrete case I am trying to style the search form with a [bootstrap field group](https://getbootstrap.com/docs/5.3/forms/input-group/#button-addons). The UI that is printed is basically a form tag plus three particles for search form, clear button and search result. Search form and clear button should be in their own "particle", so that one can design them independent from the result section. 

What I _can_ implement:

```scss
.pagefind-ui__form {
	@extend .input-group;
	@extend .mb-3;
	.pagefind-ui__search-input {
		@extend .form-control;
	}
	.pagefind-ui__search-clear {
		@extend .btn;
		@extend .btn-outline-secondary;
	}
}
```

This leads to a nice input group. But, on searching itself, this all breaks down because the no-results message appears next to the button instead of it's own section/div. 

![20230502-20:28:42-Selection-001](https://user-images.githubusercontent.com/83281/235680561-b9e281a0-67a3-4ae5-8c02-8d63f677b909.png)

Long story short: Is there a way to override the search form, result section? Is it maybe in the roadmap somewhere?

If that is a reason to use the JSApi, did anybody already invent that specific wheel in a demo script somewhere?

Maybe the "bug" is simply, that the results should not be inside of the search form, but after it?

**Comment by davidsneighbour** [comment]
I take that back. It can all be solved with more CSS instead of just Bootstrap classes. 

```scss
.pagefind-ui__drawer {
	@extend .w-100;
}
```

works for a start.
